habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -760,11 +760,27 @@ def _add_objs(
                     for motor_id in ao.existing_joint_motor_ids:
                         ao.remove_joint_motor(motor_id)
                 self.art_objs.append(ao)
+        elif self.prev_remove_recep:
+            # add prev remove recep
+            rom.add_object_by_template_handle(self.prev_remove_recep)
 
         if 'dataset' in ep_info.info and ep_info.info['dataset'] == 'mp3d':
             receps = self.set_receptacle_in_scene(ep_info)
             self._receptacles_cache[ep_info.scene_id] = receps
             self._receptacles = receps
+        elif 'dataset' in ep_info.info and ep_info.info['dataset'] == 'hssd':
+            rom = self.get_rigid_object_manager()
+            old_recep_handle = ep_info.info['old_recep_handle']
+            recep_to_remove = rom.get_object_by_handle(old_recep_handle)
+            # save the recep to move
+            move_recep_template = otm.get_templates_by_handle_substring(recep_to_remove.handle[:-6])
+            move_recep_template = list(move_recep_template.keys())[0]
+            self.prev_remove_recep = move_recep_template
+
+            rom.remove_object_by_id(recep_to_remove.object_id)
+            receps = self.set_receptacle_in_scene(ep_info, "wall_cabinet")
+            self._receptacles_cache[ep_info.scene_id] = receps
+            self._receptacles = receps
```

Comment:
Is this code used to add shelves into hssd scene?

Reply:
Yep, we were just debugging this code snippet together. @Zhouxunzhe